src/domain/prices/prices.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../logging/logging.interface';
+import { IPricesApi } from '../interfaces/prices-api.interface';
+import { IPricesRepository } from './prices.repository.interface';
+import { isArray } from 'lodash';
+
+@Injectable()
+export class PricesRepository implements IPricesRepository {
```

Comment:
In this class, I'm assuming we want to fail back to 0 (and log the error) if any error happens when getting the price. Other options considered when failing:
- Returning `null` for the price: Doesn't seem appropriate since the clients may not be handling it.
- Omitting the failed token in the balances response: doesn't seem appropriate either. 
- Throwing an error to the client: doesn't seem appropriate to fail completely when a specific price is not reachable.

Reply:
Since we are in the `domain` layer I don't agree with this exactly. I've expanded more on why in https://github.com/safe-global/safe-client-gateway/pull/732#discussion_r1350075295